pkg/zonemgr/internal/zone-file-cache/zone_file_cache.go

```diff
@@ -114,20 +114,29 @@ func buildARecordsArr(name string, namespace string, interfaces []v1.VirtualMach
 		IPs := iface.IPs
 		for _, IP := range IPs {
 			if net.IsIPv4String(IP) {
-				recordsArr = append(recordsArr, generateARecord(name, namespace, iface.Name, IP))
+				recordsArr = append(recordsArr, generateIfaceARecord(name, namespace, iface.Name, IP))
 				break
 			}
+                        defaultIP := IP
```

Comment:
Since we don't support/ever checked IPv6, I would move it to `if net.IsIPv4String(IP)`.